src/components/DatePicker/index.android.js

```diff
@@ -4,19 +4,31 @@ import moment from 'moment';
 import TextInput from '../TextInput';
 import CONST from '../../CONST';
 import {propTypes, defaultProps} from './datepickerPropTypes';
+import DateUtils from '../../libs/DateUtils';
 
 class DatePicker extends React.Component {
     constructor(props) {
         super(props);
-
+        const value = DateUtils.getDateAsText(props.value) || DateUtils.getDateAsText(props.defaultValue) || CONST.DATE.MOMENT_FORMAT_STRING;
         this.state = {
             isPickerVisible: false,
+            value,
+            selectedDate: props.value ? moment(props.value).toDate() : new Date(),
```

Comment:
I'm not sure we need to keep `value` and `selectedDate` in state, since `selectedDate` is computed based on value.

Reply:
Will do!